firestore/src/android/document_snapshot_android.cc

```diff
@@ -114,5 +114,11 @@ FieldValue DocumentSnapshotInternal::Get(const FieldPath& field,
   return FieldValueInternal::Create(env, field_value);
 }
 
+bool operator==(const DocumentSnapshotInternal& lhs,
+                const DocumentSnapshotInternal& rhs) {
+  Env env = FirestoreInternal::GetEnv();
```

Comment:
I presume both this pattern and the pattern in `common` will be repeated a lot. What do you think about creating a helper template function?

```cpp
// Defined elsewhere
template <typename T>
bool CompareJni(const T& lhs, const T& rhs) {
  Env env = FirestoreInternal::GetEnv();
  return Object::Equals(env, lhs.ToJava(), rhs.ToJava());
}

// ...
bool operator==(const DocumentSnapshotInternal& lhs,
                const DocumentSnapshotInternal& rhs) {
  return CompareJni(lhs, rhs);
}
```

And similarly for the the pattern in `document_snapshot.cc`.

Reply:
Done